src/gtpu/pktinfo.c

```diff
@@ -14,27 +14,63 @@
 #include "log.h"
 
 u64 ip4_rm_header(struct sk_buff *skb, unsigned int hdrlen){
-    struct iphdr *iph;
+    struct iphdr *iph, *inner_iph;
     struct tcphdr *tcp;
+    struct udphdr *udp;
+    struct sk_buff *inner_skb, tmp;
     u64 volume;
 
-    iph = ip_hdr(skb);
-    volume = skb->len - hdrlen - iph->ihl * 4;
-
+    volume = skb->len;
+    // hdrlen included the header length of UDP
     // Deduct the header length of transport layer
-    switch (iph->protocol) {
-    case IPPROTO_TCP:
-        // tcp = (struct tcphdr *)(skb_transport_header(skb) + (iph->ihl << 2));
-        tcp = (struct tcphdr *)(skb_transport_header(skb));
-        volume -= tcp->doff * 4;
-        break;
-    case IPPROTO_UDP:
-        volume -= 8; // udp header len = 8B
-        break;
-    default:
-        break;
-	}
 
+    udp = udp_hdr(skb);
```

Comment:
This line may be removed?  This skb is start with IP layer, or did I misunderstand something?